Source/panels/charpanel.cpp

```diff
@@ -235,87 +204,85 @@ void DrawShadowString(const Surface &out, const PanelEntry &entry)
-	LoadArt("data\\boxrightend26.pcx", &PanelParts[2]);
-	LoadArt("data\\boxleftend27.pcx", &PanelParts[3]);
-	LoadArt("data\\boxmiddle27.pcx", &PanelParts[4]);
-	LoadArt("data\\boxrightend27.pcx", &PanelParts[5]);
+	if (!CharPanelLoaded) {
+		LoadArt("data\\charbg.pcx", &PanelFull);
+		LoadArt("data\\boxleftend27.pcx", &PanelBoxLeft);
+		LoadArt("data\\boxmiddle27.pcx", &PanelBoxMiddel);
+		LoadArt("data\\boxrightend27.pcx", &PanelBoxRight);
+	}
 
 	const Surface out(PanelFull.surface.get());
 
 	for (auto &entry : panelEntries) {
-		if (entry.statDisplayFunc != nullptr) {
-			if (entry.high)
-				DrawPanelFieldHigh(out, entry.position, entry.length);
-			else
-				DrawPanelFieldLow(out, entry.position, entry.length);
+		if (!CharPanelLoaded && entry.statDisplayFunc != nullptr) {
+			DrawPanelField(out, entry.position, entry.length);
 		}
 		DrawShadowString(out, entry);
 	}
 
-	for (auto &gfx : PanelParts) {
-		gfx.Unload();
-	}
+	PanelBoxLeft.Unload();
+	PanelBoxMiddel.Unload();
+	PanelBoxRight.Unload();
 }
 
-bool CharPanelLoaded = false;
-
 void DrawStatButtons(const Surface &out)
 {
 	if (MyPlayer->_pStatPts > 0) {
 		if (MyPlayer->_pBaseStr < MyPlayer->GetMaximumAttributeValue(CharacterAttribute::Strength))
-			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 159 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Strength)] ? 3 : 2);
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 157 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Strength)] ? 3 : 2);
 		if (MyPlayer->_pBaseMag < MyPlayer->GetMaximumAttributeValue(CharacterAttribute::Magic))
-			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 187 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Magic)] ? 5 : 4);
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 185 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Magic)] ? 5 : 4);
 		if (MyPlayer->_pBaseDex < MyPlayer->GetMaximumAttributeValue(CharacterAttribute::Dexterity))
-			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 216 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Dexterity)] ? 7 : 6);
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 214 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Dexterity)] ? 7 : 6);
 		if (MyPlayer->_pBaseVit < MyPlayer->GetMaximumAttributeValue(CharacterAttribute::Vitality))
-			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 244 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Vitality)] ? 9 : 8);
+			CelDrawTo(out, GetPanelPosition(UiPanels::Character, { 137, 242 }), *pChrButtons, chrbtn[static_cast<size_t>(CharacterAttribute::Vitality)] ? 9 : 8);
 	}
 }
 
 } // namespace
 
 void DrawChr(const Surface &out)
 {
-	if (!CharPanelLoaded) {
-		LoadCharPanel();
-		CharPanelLoaded = true;
-	}
+	//if (!CharPanelLoaded) {
+	LoadCharPanel();
+	CharPanelLoaded = true;
+	//}

```

Comment:
```suggestion
	LoadCharPanel();
```

Reply:
This will be relevant again once i get TRN working, I have it coded but it's crashing...

Reply:
I'll await your further changes then.